CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift

```diff
@@ -0,0 +1,454 @@
+//
+//  CardPartProgressBarView.swift
+//  CardParts
+//
+//  Created by Venkatnarayansetty, Badarinath on 9/19/19.
+//
+
+import Foundation
+import UIKit
+import RxSwift
+import RxCocoa
+
+public class CardPartProgressBarView: UIView, CardPartView {
+    
+    public var margins: UIEdgeInsets = CardParts.theme.cardPartMargins
+    
+    public var animatable: Bool = false
+    
+    private var markerView: UIView!
+    
+    private var markerLabel: UILabel!
+    
+    private var valueBarsStackView: UIStackView!
+    
+    private var colorBarsStackView: UIStackView!
+    
+    private var colorBarsStackViewContainerView: UIView!
+    
+    private var showButtomValues: Bool = false
+    
+    private var totalHeight: CGFloat = 0
+    
+    private var progressBarOnlyHeight: CGFloat = 10
+    
+    private var markerHeight: CGFloat = 7
+    
+    private var valuesBarHeight: CGFloat = 0
+    
+    private var markerLabelViewHeight: CGFloat = 0
+    
+    fileprivate var currentVal: Double = 0
+    
+    
+    public init(barValues:[Double], barColors: [UIColor], marker: UIView? = nil , markerLabelTitle: String, currentValue: Double, showShowBarValues: Bool) {
+        
+        self.barColors = barColors
```

Comment:
Can `barValues` and `barColors` be of different sizes? If not, can we protect against that?

Reply:
It's customizable based on the values u provide. `   let barValues: [Double] = [300, 600, 650, 700, 750, 850]`